runtime/service.go

```diff
@@ -884,6 +891,39 @@ func (s *service) newIOProxy(logger *logrus.Entry, stdin, stdout, stderr string,
 	return ioConnectorSet, nil
 }
 
+func (s *service) addFIFOs(taskID, execID string, config cio.Config) error {
+	s.fifosMu.Lock()
+	defer s.fifosMu.Unlock()
+
+	_, exists := s.fifos[taskID]
+	if !exists {
+		s.fifos[taskID] = make(map[string]cio.Config)
+	}
+
+	value, exists := s.fifos[taskID][execID]
+	if exists {
+		return fmt.Errorf("(task=%q, exec=%q) must not have FIFOs, but there was %+v", taskID, execID, value)
```

Comment:
I think this could be reworded to be more clear. Why must we not have FIFOs already set here? (because this would overwrite them?)

Reply:
> Why must we not have FIFOs already set here? (because this would overwrite them?)

Yeah. I don't want to overwrite the FIFO files.